clients/go/pkg/worker/jobStreamer.go

```diff
@@ -0,0 +1,116 @@
+// Copyright © 2018 Camunda Services GmbH (info@camunda.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package worker
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/camunda/zeebe/clients/go/v8/pkg/commands"
+)
+
+type jobStreamer struct {
+	request         commands.DispatchStreamJobsCommand
+	workerFinished  chan bool
+	closeSignal     chan struct{}
+	backoffSupplier BackoffSupplier
+	retryDelay      time.Duration
+
+	closedMutex sync.Mutex
+	closed      bool
+
+	streamMutex sync.Mutex
+}
+
+func (streamer *jobStreamer) stream(closeWait *sync.WaitGroup) {
```

Comment:
❓ Just to clarify the job streamer is just to handle the stream lifecycle? How does the consumptions of jobs works, this is somewhere else? 🤔

Reply:
Exactly. Streaming jobs and so on is entirely handled by the `StreamJobsCommand`, which dumps them in the `jobQueue`. The `jobDispatcher.go` then takes care of passing whatever from that channel to the worker go routines.